Extension/src/LanguageServer/client.ts

```diff
@@ -1294,20 +1299,29 @@ export class DefaultClient implements Client {
         this.rootFolder = workspaceFolder;
         this.rootRealPath = this.RootPath ? (fs.existsSync(this.RootPath) ? fs.realpathSync(this.RootPath) : this.RootPath) : "";
 
-        let storagePath: string | undefined;
-        if (util.extensionContext) {
-            const path: string | undefined = util.extensionContext.storageUri?.fsPath;
-            if (path) {
-                storagePath = path;
-            }
+        this.baseStoragePath = util.getBaseStoragePath();
+        this.legacyStoragePath =  "";
+        let storagePath: string = "";
+        let workspaceHash: string = "";
+
+        this.legacyStoragePath = util.extensionContext?.storageUri?.fsPath ?? "";
+        if (this.legacyStoragePath.length > 0) {
+            workspaceHash = path.basename(path.dirname(this.legacyStoragePath));
         }
 
-        if (!storagePath) {
-            storagePath = this.RootPath ? path.join(this.RootPath, "/.vscode") : "";
+        if (this.legacyStoragePath.length > 0 ) {
+            this.legacyStoragePath = this.RootPath ? path.join(this.RootPath, ".vscode") : "";
         }
+
         if (workspaceFolder && vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1) {
-            storagePath = path.join(storagePath, util.getUniqueWorkspaceStorageName(workspaceFolder));
+            this.legacyStoragePath = path.join(this.legacyStoragePath, util.getUniqueWorkspaceStorageName(workspaceFolder));
         }
+
+
+        if ((this.baseStoragePath?.length !== undefined) && (workspaceHash.length > 0)) {
```

Comment:
Why is this checking for length is undefined `(this.baseStoragePath?.length !== undefined)`? Is length only a number?